rpm/generic/zfs.spec.in

```diff
@@ -167,6 +167,21 @@ for managing zpools
 %postun -n libzpool5 -p /sbin/ldconfig
 %endif
 
+%package -n libzdb1
+Summary:        ZDB library
+Group:          System Environment/Kernel
+
+%description -n libzdb1
+This package contains the zdb library, which is an unstable
+library exposing the innards of zdb.
+
+%if %{defined ldconfig_scriptlets}
+%ldconfig_scriptlets -n libzdb1
+%else
+%post -n libzdb1 -p /sbin/ldconfig
+%postun -n libzdb1 -p /sbin/ldconfig
+%endif
```

Comment:
This doesn't go quite far enough.  If we're going to split `libzdb` in to its own sub-package we'll also need to create a `libzfs1-devel` package and add the needed `Requires` and `BuildRequires` dependencies to the other rpm packages.  I'm sure there's some equivalent needed for the Debian packaging as well.

Or we could build this as a convenience library which is only used internally.

Either way there's a nice library dependency graph in `lib/Makefile.am` we'll need to update to reflect this change.

```
#
# Shown below is a simplified dependency graph of the OpenZFS provided
# libraries.  Administrative commands (`zfs`, `zpool`, etc) interface with
# the kernel modules using the `libzfs.so` and `libzfs_core.so` libraries.
# These libraries provide a stable ABI across OpenZFS point releases.
#
# The `libzpool.so` library is a user space build of the DMU and SPA layers
# used to implement debugging tools (zdb) and code validation tools (ztest).
# These library interfaces are subject to change at any time.
#
#
# CMDS:   zhack/ztest/zdb/            zfs/zpool/zed/
#        raidz_{test,bench}          zinject/zstream
#                |                          |
# LIBS:          |                          |              libzfsbootenv*
#                |                          |                    |
#                |                          |                    |
#             libzpool                   libzfs* ----------------+
#             | | |  \                  / | | |
#    libicp --/ | |   \                /  | | \------- libshare
#               | |    \              /   | |
#    libzstd ---/ |     \            /    | \--------- libuutil
#                 |      \          /     \              | |
#    libunicode --/       \        /       \             | |
#                          \      /         \            | |
#                          libzutil        libzfs_core*  | |
#                          | | | | \          |     |    | |
#                          | | | | |          |     |    | |
#                          | | | | |          |     |    | |
#    libtpool -------------/ | | | \---- libnvpair* |    | |
#                            | | |                  |    | |
#    libefi -----------------/ | \------ libavl* --------/ |
#                              |                    |      |
#                              \-------- libspl ----+------/
#
#
# NB: GNU Automake Manual, Chapter 8.3.5: Libtool Convenience Libraries
# These nine libraries are intermediary build components.
#
# * - A stable ABI is provided for these libraries;
# when performing an ABI check the following options are applied:
#
# --no-unreferenced-symbols: Exclude symbols which are not referenced by
# any debug information.  Without this _init() and _fini() are incorrectly
# reported on CentOS7 for libuutil.so.
#
# --headers-dir1: Limit ABI checks to public OpenZFS headers, otherwise
# changes in public system headers are also reported.
#
# --suppressions: Honor a suppressions file for each library to provide
# a mechanism for suppressing harmless warnings.
#
```

Reply:
I don't actually want it in a sub-package, I thought I dropped that change, it was just there because the rpm build was failing and I was trying to get it to shut up.

I don't really want anyone external depending on this anytime soon.

Also, I'm not touching the native Debian packaging, if the people who contributed it won't fix the most trivial of bugs in it, I'm probably going to open a PR to remove it, since "make -j no longer works" is kind of absurd.